ha/setup/create_pacemaker_resources.py

```diff
@@ -19,262 +19,289 @@
+        op start timeout=60s interval=0s \
+        op monitor timeout=30s interval=30s \
+        op stop timeout=60s interval=0s --group management_group")
     if push:
         cib_push(cib_xml)
 
+def uds(cib_xml, push=False, **kwargs):
+    """Create uds resource."""
+    with_uds = kwargs["uds"] if "uds" in kwargs else False
+    if with_uds:
+        process.run_cmd(f"pcs -f {cib_xml} resource create uds systemd:uds \
+            op start timeout=60s interval=0s \
+            op monitor timeout=30s interval=30s \
+            op stop timeout=60s interval=0s")
+        if push:
+            cib_push(cib_xml)
+
+def config_constraint(cib_xml, push=False, **kwargs):
+    """
+    Configure all constaints
 
-def uds(cib_xml, push=False):
-    """Create uds resource and constraints."""
-    cmd_uds = f"pcs -f {cib_xml} resource create uds systemd:uds op monitor interval=30s"
-    SimpleCommand().run_cmd(cmd_uds)
+    Args:
+        cib_xml (str): cib cluster file.
+    """
     constraints = [
-            f"pcs -f {cib_xml} constraint colocation add uds with csm-agent score=INFINITY",
+            f"pcs -f {cib_xml} constraint order io_group-clone then motr-free-space-mon",
+            f"pcs -f {cib_xml} constraint colocation add motr-free-space-mon with io_group-clone",
+            f"pcs -f {cib_xml} constraint colocation add s3backprod with io_group-clone"
             # According to EOS-9258, there is a bug which requires UDS to be started after csm_agent
+            f"pcs -f {cib_xml} constraint colocation add uds with csm-agent score=INFINITY",
             f"pcs -f {cib_xml} constraint order csm-agent then uds"
-            ]
+        ]
+    with_uds = kwargs["uds"] if "uds" in kwargs else False
     for c in constraints:
-        SimpleCommand().run_cmd(c)
+        if with_uds == False and "uds" in c:
+            continue
+        process.run_cmd(c)
     if push:
         cib_push(cib_xml)
 
+core_io = [hax, motr, haproxy, s3auth, s3servers]
+io_helper_aa = [s3bc]
+io_helper_ap = [free_space_monitor, s3bp]
+monitor_config = [sspl]
+management_config = [mgmt_vip, kibana, csm, uds]
+# TODO: ha_group
 
-def mgmt_stack(cib_xml, mgmt_vip_cfg, with_uds=False, push=False):
-    """Create Mgmt stack related resources.
-
-    It also creates and defines management group to support colocation and
-    ordering requirements.
-    """
-    mgmt_resources(cib_xml)
-
-    if with_uds:
-        uds(cib_xml)
-
-    cmd = f"pcs -f {cib_xml} resource group add management kibana csm-agent csm-web"
-    SimpleCommand().run_cmd(cmd)
-
-    if mgmt_vip_cfg:
-        mgmt_vip(cib_xml, **mgmt_vip_cfg, push=False)
-        cmd_group = f"pcs -f {cib_xml} resource group add management kibana --before csm-agent"
-        SimpleCommand().run_cmd(cmd_group)
+def io_stack(cib_xml, push, **kwargs):
+    """Create IO stack related resources."""
+    # Create core io resources
+    for fun in core_io:
```

Comment:
Please replace "fun" by some other name, may be create_resource.

Reply:
Done